### PR TITLE
Toxic Exposure: Handle dates with XX or month or day

### PR DESCRIPTION
**Pull request by sethdarragile6**
During the 10% rollout of TE, a few submissions came through with XXs for the month or day (indicating partial days), and failed when trying to parse them into an LH request. The FE seems to effectively validate against this and prevent submission, so the root cause is still a mystery. In the meantime, this BE fix will handle these dates accordingly.

## Summary

- *This work is behind a feature toggle (flipper): YES*
- *(Summarize the changes that have been made to the platform)*
Method that parses incoming TE dates modified to account for XX in date parts
- *(If bug, how to reproduce)*
- *(What is the solution, why is this the solution?)*
- *(Which team do you work for, does your team own the maintenance of this component?)*
Disability Benefits Experience Team 1 - DBEX TREX 🦖
- *(If introducing a flipper, what is the success criteria being targeted?)*

## Related issue(s)

- #[92417](https://app.zenhub.com/workspaces/disability-benefits-experience-team-1-63dbdb0a401c4400119d3a44/issues/gh/department-of-veterans-affairs/va.gov-team/92417)
- *Link to previous change of the code/bug (if applicable)*
- *Link to epic if not included in ticket*

## Testing done

- [x] *New code is covered by unit tests*
- *Describe what the old behavior was prior to the change*
- *Describe the steps required to verify your changes are working as expected. Exclusively stating 'Specs run' is NOT acceptable as appropriate testing*
- *If this work is behind a flipper:*
  - *Tests need to be written for both the flipper on and flipper off scenarios. [Docs](https://depo-platform-documentation.scrollhelp.site/developer-docs/feature-toggles-guide#Featuretogglesguide-Backendexample).*
  - *What is the testing plan for rolling out the feature?*

## Screenshots
_Note: Optional_

## What areas of the site does it impact?
*(Describe what parts of the site are impacted and*if*code touched other areas)*

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: lib/evss/disability_compensation_form/form526_to_lighthouse_transform.rb
Comment (kylesoskin) [inline lib/evss/disability_compensation_form/form526_to_lighthouse_transform.rb:687]: Not sure if you've seen any instances of it, but Ive seen like `2019-03` before I think, like to mean sometime in march of 2019. 

```
irb#1(main):031> convert_date_no_day("2019-02")
(irb#1):11:in `parse': invalid date (Date::Error)
	from (irb#1):11:in `convert_date_no_day'
	from (irb#1):31:in `<top (required)>'
```

Not sure if it is worth adding in to account for that. if we arent seeing that, probably doesnt matter... I just thought I recalled seeing that, in the past, and it is not much more effort to support.

File: lib/evss/disability_compensation_form/form526_to_lighthouse_transform.rb
Comment (aurora-a-k-a-lightning) [inline lib/evss/disability_compensation_form/form526_to_lighthouse_transform.rb:686]: should we normalize these "XX"s - either `.downcase` or `.upcase` the `month`/`day` variables?

**Review by aurora-a-k-a-lightning — APPROVED** [review]
**_this looks great!_**
agreed with kyle - we could prevent some terrible things from happening by normalizing the XXs, and accounting for the missing `day` for the `Date.parse(%Y-%m)` that will cause errors.

but i will leave that to your discretion as this works for the problem we were seeing for the release as-is!

File: lib/evss/disability_compensation_form/form526_to_lighthouse_transform.rb
Comment (sethdarragile6) [reply lib/evss/disability_compensation_form/form526_to_lighthouse_transform.rb:687]: Haven't seen that yet, but it was easy to support (as you said) so I added that check.

File: lib/evss/disability_compensation_form/form526_to_lighthouse_transform.rb
Comment (sethdarragile6) [reply lib/evss/disability_compensation_form/form526_to_lighthouse_transform.rb]: @aurora-a-k-a-lightning  thanks! I incorporated Kyle's suggestion and the upcase one. 

@christinec-fftc I think I'll hold off on accounting for the XXXX year for now, unless we actually start seeing this. If a date is XXXX-XX-XX, I would think (hope) it's coming in as nil or not at all. Otherwise there could be things like XXXX-01-XX or XXXX-XX-01 which again seem like they would be bizarre and rare. I may have to eat these word down the line, but for now I'd like to keep it simple and target the known scenario.

**Review by kylesoskin — APPROVED** [review]
Looks good!

**Review by tommasina-va — APPROVED** [review]
Great to know this will catch any of those sneaky 'xx's' ! 🚢

**Review by aurora-a-k-a-lightning — APPROVED** [review]

**Review by ryan-mcneil — APPROVED** [review]
👍

**Pull request merged**
